provisioning/service/src/Contract/ContractApiHttp.cs

```diff
@@ -134,8 +135,8 @@ internal class ContractApiHttp : IContractApiHttp
                     using HttpResponseMessage httpResponse = await _httpClientObj.SendAsync(msg, cancellationToken).ConfigureAwait(false);
                     if (httpResponse == null)
                     {
-                        throw new ProvisioningServiceClientTransportException(
-                            $"The response message was null when executing operation {httpMethod}.");
+                        throw new DeviceProvisioningServiceException(
+                            $"The response message was null when executing operation {httpMethod}.", innerException: null, isTransient: true);
```

Comment:
FWIW, if you name the parameters, you don't need to include the defaults. So, this could probably be:

```suggestion
                            $"The response message was null when executing operation {httpMethod}.", isTransient: true);
```